src/main.rs

```diff
@@ -330,15 +332,11 @@ fn run(args: Cli) -> Result<i32> {
         s.save_headers(&headers)?;
 
         let mut cookie_jar = cookie_jar.lock().unwrap();
-        for cookie in s.cookies() {
-            match cookie_jar.insert_raw(&cookie, &url) {
-                Ok(..) | Err(cookie_store::CookieError::Expired) => {}
-                Err(err) => return Err(err.into()),
-            }
-        }
+        *cookie_jar = CookieStore::from_cookies(s.cookies(), false)?;
```

Comment:
I'm unsure if we should abort on invalid cookies. It's probably fine though. I want to compare this to HTTPie's behavior when I do a functional test.

It's weird that `CookieStore::from_cookies()` takes an iterator of `Result`s.

Reply:
I generally find the `cookie_store` crate hard to work with. I think it would benefit from a breaking change release that revisits its current API.

> I want to compare this to HTTPie's behavior when I do a functional test.

Sounds good. I have compared the unhappy path between HTTPie and xh yet.

Reply:
An unparsable `"domain"` crashes xh but not HTTPie. That seems acceptable, though we could maybe turn it into a warning and a skip.

This is what the crash looks like though:
```
xh: error: empty host
```
We probably need a `.context()` call somewhere.

Reply:
I somehow missed your last comment about unparsable "domain" errors 😬. 

>...though we could maybe turn it into a warning and a skip.

My initial approach was to filter out any `err` values from `s.cookies()` but that caused the cookie to be discarded from the session file. In the end, I went with just adding context to cookie errors.

```
xh: error: Failed to load the cookies from the session file

Caused by:
    empty host
```

Reply:
Looks good! I'll try to give this PR a final look tomorrow.